test/test_jiterator.py

```diff
@@ -0,0 +1,168 @@
+# Owner(s): ["module: cuda"]
+
+import torch
+from torch.cuda.jiterator import create_jit_fn
+import sys
+import unittest
+from torch.testing._internal.common_utils import TestCase, parametrize, run_tests, instantiate_parametrized_tests, TEST_WITH_ROCM
+
+TEST_CUDA = torch.cuda.is_available()
+
+if not TEST_CUDA:
+    print('CUDA not available, skipping tests', file=sys.stderr)
+    TestCase = object  # noqa: F811
+
+if TEST_WITH_ROCM:
+    print('Jiterator is not supported on ROCm, skipping tests', file=sys.stderr)
+    TestCase = object  # noqa: F811
+
+
+code_string = "template <typename T> T my_fused_kernel(T x, T y, T alpha, T beta) { return  -x * y + x - y + alpha - beta; }"
+jitted_fn = create_jit_fn(code_string, alpha=0, beta=0)
+
+def ref_fn(x, y, alpha=0, beta=0):
+    return -x * y + x - y + alpha - beta
+
+class TestPythonJiterator(TestCase):
+    @parametrize("dtype", [
+        torch.float, torch.double, torch.half,
+        # torch.bfloat16,  failing due to numerical difference
+        torch.uint8, torch.int8, torch.int16, torch.int, torch.long,
+        torch.complex64, torch.complex128,
+        # torch.bool,
+    ])
+    def test_all_dtypes(self, dtype):
+        a = torch.rand(3, device='cuda').mul(10).type(dtype)
+        b = torch.rand(3, device='cuda').mul(10).type(dtype)
+
+        expected = ref_fn(a, b)
+        result = jitted_fn(a, b)
+
+        rtol = 0.00001
+        if dtype is torch.half:
+            rtol = 1e-2
+        assert torch.allclose(expected, result, rtol=rtol)
```

Comment:
Use self.assertEqual (or, if you want more control, torch.testing.assert_close)

Reply:
ok.

Reply:
fixed.